frontend/packages/knative-plugin/src/components/test-function/modal-components/RequestBody.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { FormikProps, FormikValues } from 'formik/dist/types';
+import { YAMLEditorField } from '@console/shared/src';
+import { InvokeFormat } from '../../../types';
+
+const RequestBody: React.FC<FormikProps<FormikValues>> = ({ values }) => {
+  const {
+    request: {
+      headers: {
+        [InvokeFormat.CloudEvent]: { ContentType: CloudEventContentType },
+        [InvokeFormat.HTTP]: { ContentType: HTTPContentType },
+      },
+      format: invokeFormat,
+    },
+  } = values;
+
+  const getcurrentLanguage = () => {
+    if (invokeFormat === InvokeFormat.HTTP && HTTPContentType === 'application/json') {
+      return 'json';
+    }
+    if (invokeFormat === InvokeFormat.CloudEvent && CloudEventContentType === 'application/json') {
+      return 'json';
+    }
+    return 'yaml';
+  };
```

Comment:
Does the editor also supports "text" for non json and non yaml inputs?

```suggestion
  const getcurrentLanguage = () => {
    // Content type is `application/json` by default,
    // but user might add an encoding like `application/json; charset=utf-8`
    if (contentType.startsWith('application/json')) {
      return 'json';
    }
    if (contentType.startsWith('text/yaml')) {
      return 'yaml';
    }
    return 'text';
  };
```